native/spark-expr/src/conversion_funcs/cast.rs

```diff
@@ -988,6 +988,9 @@ fn is_datafusion_spark_compatible(
         return true;
     }
     match from_type {
+        DataType::Null => {
+            matches!(to_type, DataType::List(_))
+        }
```

Comment:
This was needed to fix test failures in `CometArrayExpressionSuite`.

Reply:
No, we will likely need to add support for more casts from `Null` to other types in the future, especially once we add support for more complex types.